src/main/scala/scalaparsers/Token.scala

```diff
@@ -5,21 +5,20 @@ package scalaparsers
   * @author EAK
   */
 
-sealed abstract class Token
+sealed abstract class Token extends Product with Serializable
 
-case object VSemi extends Token {
+final case object VSemi extends Token {
```

Comment:
I like the final case classes.  Is "final case object" different from "case object" in some subtle way I've not yet discovered?

Reply:
No clue, I just have a habit/compulsion to tack `final` onto any `case {class,object}`. People seem to like it https://twitter.com/viktorklang/status/735556776989491201 😉

Reply:
Yeah, I think it's great on case class.  I don't think it matters one way or another on case object.  Mostly asking out of curiosity.